akka-stream-tests/src/test/scala/akka/stream/scaladsl/QueueSinkSpec.scala

```diff
@@ -112,5 +114,26 @@ class QueueSinkSpec extends AkkaSpec {
       queue.pull().onFailure { case e ⇒ e.isInstanceOf[IllegalStateException] should ===(true) }
     }
 
+    "keep on sending even after the buffer has been full" in assertAllStagesStopped {
+      val (probe, queue) = Source(1 to 20)
+        .alsoToMat(Flow[Int].take(16).watchTermination()(Keep.right).to(Sink.ignore))(Keep.right)
+        .toMat(Sink.queue())(Keep.both)
+        .run()
+
+      // wait for the buffer to fill up before pulling
+      akka.pattern.after(0.5 seconds, system.scheduler)(queue.pull()) pipeTo testActor
```

Comment:
No, maybe that didn't help much. Having a happy path which always requires a wait for 0.5 seconds isn't very desirable. We would want it to pass as soon as the requirement has been fulfilled if possible, and then in the worst case it can wait all that time.

Something like this would therefore be better:
```Scala
within(0.5.seconds) {
  queue.pull() pipeTo testActor
  expectMsg(Some(1))
}
```